src/render/graph.rs

```diff
@@ -26,9 +29,9 @@ pub struct Node {
     /// Renderer: converts inputs to outputs
     processor: Box<dyn AudioProcessor>,
     /// Reusable input buffers
-    inputs: Vec<AudioRenderQuantum>,
+    inputs: SmallVec<[AudioRenderQuantum; 2]>,
     /// Reusable output buffers, consumed by subsequent Nodes in this graph
-    outputs: Vec<AudioRenderQuantum>,
+    outputs: SmallVec<[AudioRenderQuantum; 2]>,
```

Comment:
These are not real solutions to the deallocation problems of this struct

Reply:
Actually, maybe I'm missing something but it seems to me that the only nodes that have multiple inputs / outputs are the `ChannelSplitterNode` and `ChannelMergerNode` which are clamped to `MAX_CHANNELS`, cf. https://webaudio.github.io/web-audio-api/#dom-baseaudiocontext-createchannelmerger, no?

Then maybe we could simply use an `ArrayVec<AudioRenderQuantum, MAX_CHANNELS>` to fix this?

Reply:
~~True that, but of course we also allow users to implement their own AudioNodes which I think could use an unbounded number of inputs/outputs.~~

Scratch that, in the JS interface the https://webaudio.github.io/web-audio-api/#AudioWorkletNode is bound to 1 input and 1 output. So technically I could restrict the raw AudioNode trait and panic when the users supplies a 32+ input/output count.

We need to check performance though, `ArrayVec<AudioRenderQuantum, MAX_CHANNELS>` is a rather large object to have on the stack, and rust will probably memcpy it around a lot, so we may want to have it on the heap anyway. We can experiment with pre-allocating that `Vec` on the control thread